### ISSUE TITLE
Mongodb always uses "test" as database name since 0.2.25

### ISSUE THREAD

**Issue by Lectem** [typeorm/typeorm#6157]
**Issue type:**

[ ] question
[x] bug report
[ ] feature request
[ ] documentation issue

**Database system/driver:**

[ ] `cordova`
[x] `mongodb`
[ ] `mssql`
[ ] `mysql` / `mariadb`
[ ] `oracle`
[ ] `postgres`
[ ] `cockroachdb`
[ ] `sqlite`
[ ] `sqljs`
[ ] `react-native`
[ ] `expo`

**TypeORM version:**

[x] `latest` (0.2.25)
[ ] `@next`
[ ] `0.x.x` (or put your version here)

**Steps to reproduce or a small repository showing the problem:**

When updating to the 0.2.25 (from 0.2.24), _typeorm_ now always uses the database named 'test' instead of what is passed in the options when using mongodb.

Downgrading to 0.2.24 makes it work again.

We are using `"@nestjs/typeorm": "^7.0.0"`.

With the debugger we can see that the option is correct:

![image](https://user-images.githubusercontent.com/6024225/83271171-c2632f80-a1c9-11ea-8187-c85f87bdbfe1.png)

From what I can see, it's simply always using the default database name, taken from `node_modules\mongodb\lib\core\uri_parser.js`:

line 673:

```
  if (result.auth && result.auth.db) {
    result.defaultDatabase = result.auth.db;
  } else {
    result.defaultDatabase = 'test';
  }
```

replacing test with another value will give me that value as database name.

Version 0.2.24 has no such issue.

**Comment by imnotjames** [comment]
Is this still an issue?

**Comment by Lectem** [comment]
Yes, I can repro when updating to 0.2.25.
I wish I had a minimal repro but sadly don't have much time right now. :disappointed: 
I'm guessing that any nestjs application using mongodb should trigger this though.

**Comment by Lectem** [comment]
Seems related to #5676.
Since the url does not contain the database, the call to `buildDriverOptions` will overwrite it (see https://github.com/typeorm/typeorm/blob/54a3a15c4c1afc9a2cc42103d4498847f67820f9/src/driver/DriverUtils.ts#L31).
So options which contained 

![image](https://user-images.githubusercontent.com/6024225/91306839-6babc880-e7ad-11ea-8ee9-e34f1661d67e.png)

Gets replaced by `urlDriverOptions` which contains this:

![image](https://user-images.githubusercontent.com/6024225/91306933-9138d200-e7ad-11ea-8d2f-4d1cf7927102.png)

And so in the end, `this.driver.database` is undefined, while in 0.2.24 it had the correct value (in the above example, 'coucou').

From what I saw it is consistent with https://github.com/typeorm/typeorm/blob/d1ed5723bb55d4124be13b21e57d9f729d0bae1a/src/driver/mongodb/MongoDriver.ts#L429 but this is very un-intuitive that you can not mix a url with more options, especially since it used to work just fine in 0.2.24, as it seems `this.driver.database` is in fact what is used at the end of the day when not `undefined`.

Also note that it isn't documented anywhere that `options.database` will not be taken into account if you provide a `options.url`.

**Comment by imnotjames** [comment]
I think that 0.2.26 no longer has this issue - I'm having trouble reproducing in that version and the line of code that was a problem -- the one you see in your screenshot -- is actually different from what's in 0.2.26. It was a bug that was introduced & was fixed for 0.2.26.

Does this still occur with 0.2.26?

**Comment by imnotjames** [comment]
Let me know if the issue persists - however, as I can't replicate it with 0.2.26 & 0.2.27 I think we're good to go here!
